### PR TITLE
Fix section on creating new rails environment by providing more information on what should be done

### PR DESCRIPTION
**Pull request by ibrahimab**
### Summary

Current rails guide did not provide enough information on how to create a new rails env

### Suggestions (which I would like to implement)
- rails generator for creating a new environment
- refactoring environment specific js configuration like `babel.config.js` to a new config file

### CODE REVIEW

File: guides/source/configuring.md [hunk 1]
```diff
@@ -1410,7 +1410,20 @@ development:
...
 
-Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To define an environment called "staging" for this server, just create a file called `config/environments/staging.rb`. Please use the contents of any existing file in `config/environments` as a starting point and make the necessary changes from there.
+Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To create this environment the following actions are required:
+
+- create a new `config/environments/staging.rb` file
+- add a new database configuration entry in `config/database.yml`
+- generate a new secret key, for which you can use `bin/rails secret`
+- generate new credentials using `bin/rails credentials:edit --environment=staging` and add your newly generated secret under `secret_key_base`
+
+These actions are required if you have webpack:
```
Comment (jacobherrington) [inline guides/source/configuring.md:1420]:
```suggestion
For applications using webpack:
```

I think the intent is a bit more clear if we do this, and it removes the use of "you." I don't think it is a hard rule, but it's recommended in the API docs contributing guidelines (maybe the guides are even more relaxed about this).

File: guides/source/configuring.md [hunk 2]
```diff
@@ -1410,7 +1410,20 @@ development:
...
+- edit `babel.config.js` to include your environment specific checks and add `staging` to the `validEnv` array
+- create a new `config/webpack/staging.js` file, see `config/webpack/production.js` for an example
+- edit `config/webpacker.yml` and include a `staging` entry
+
+This action is required if you have actioncable:
```
Comment (jacobherrington) [inline guides/source/configuring.md:1425]:
```suggestion
For applications using Action Cable:
```

This suggestion is based on the same reasoning as above. Also, other guides seem to prefer "Action Cable" to "actioncable" or "ActionCable," we should probably try to remain consistent. :smile:

File: guides/source/configuring.md [hunk 3]
```diff
@@ -1410,7 +1410,20 @@ development:
...
 By default Rails ships with three environments: "development", "test", and "production". While these are sufficient for most use cases, there are circumstances when you want more environments.
 
-Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To define an environment called "staging" for this server, just create a file called `config/environments/staging.rb`. Please use the contents of any existing file in `config/environments` as a starting point and make the necessary changes from there.
+Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To create this environment the following actions are required:
+
+- create a new `config/environments/staging.rb` file
+- add a new database configuration entry in `config/database.yml`
+- generate a new secret key, for which you can use `bin/rails secret`
+- generate new credentials using `bin/rails credentials:edit --environment=staging` and add your newly generated secret under `secret_key_base`
+
```
Comment (jacobherrington) [inline guides/source/configuring.md:1419]:
```suggestion
* Create a new `config/environments/staging.rb` file.
* Add a new database configuration entry in `config/database.yml`.
* Generate a new secret key using `bin/rails secret`.
* Generate new credentials using `bin/rails credentials:edit --environment=staging`, and add your newly generated secret under `secret_key_base`.

```

This guide uses bullets elsewhere with a different format, I think it makes sense to remain consistent? Do you agree?

File: guides/source/configuring.md [hunk 4]
```diff
@@ -1410,7 +1410,20 @@ development:
...
+These actions are required if you have webpack:
+- edit `babel.config.js` to include your environment specific checks and add `staging` to the `validEnv` array
+- create a new `config/webpack/staging.js` file, see `config/webpack/production.js` for an example
+- edit `config/webpacker.yml` and include a `staging` entry
```
Comment (jacobherrington) [inline guides/source/configuring.md:1423]:
```suggestion
* Edit `babel.config.js` to include your environment-specific checks, and add `staging` to the `validEnv` array.
* Create a new `config/webpack/staging.js` file, see `config/webpack/production.js` for an example.
* Edit `config/webpacker.yml` and include a `staging` entry.
```

File: guides/source/configuring.md [hunk 5]
```diff
@@ -1410,7 +1410,20 @@ development:
...
+
+This action is required if you have actioncable:
+- edit `config/cable.yml` and include a `staging` entry and be sure to also provide the `channel_prefix` key using the format `<app_name>_<environment>`
```
Comment (jacobherrington) [inline guides/source/configuring.md:1426]:
```suggestion
* Edit `config/cable.yml` and include a `staging` entry, and be sure to also provide the `channel_prefix` key using the format `<app_name>_<environment>`.
```

File: guides/source/configuring.md [hunk 6]
```diff
@@ -1410,7 +1410,20 @@ development:
 
 By default Rails ships with three environments: "development", "test", and "production". While these are sufficient for most use cases, there are circumstances when you want more environments.
 
-Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To define an environment called "staging" for this server, just create a file called `config/environments/staging.rb`. Please use the contents of any existing file in `config/environments` as a starting point and make the necessary changes from there.
+Imagine you have a server which mirrors the production environment but is only used for testing. Such a server is commonly called a "staging server". To create this environment the following actions are required:
```
Comment (jacobherrington) [inline guides/source/configuring.md:1413]:
```suggestion
Imagine you have a server mirroring the production environment, but is only used for testing. Such a server is commonly called a "staging server". To create a staging environment perform the following actions:
```
I added a comma, but I also added some minor wording suggestions. What do you think?

**Review by jacobherrington — COMMENTED** [review]
This is a great thing to document IMO.

I added some comments, mostly regarding formatting. You're welcome to commit them if you agree! :smile:

**Comment by ibrahimab** [comment]
Great improvements, I have committed your suggestions!

**Comment by ibrahimab** [comment]
Is there any update on the review for this?

**Comment by jonathanhefner** [comment]
>     * rails generator for creating a new environment

I feel like that would be a much friendlier and more Railsy solution.  And unit tests could verify that all necessary steps are included.  Would you like to submit a PR for that?

**Comment by ibrahimab** [comment]
I'll get started on it right away.

**Comment by rafaelfranca** [comment]
Thank for the pull request. New Rails environment are an anti-pattern in my opinion and Rails should not be advertising it. Heroku has a good explanation why people should be avoiding creating new Rails environments https://devcenter.heroku.com/articles/deploying-to-a-custom-rails-environment.

**Comment by ibrahimab** [comment]
It's weird to promote setting up a staging environment using ENV variables rather than using the environment structure Rails already has. Rails philosophy has always been convention over configuration, and using the same environment that can be changed with ENV does not look like convention, that looks like hacking/monkey patching which seems more of an anti pattern to me. And closing an issue without a discussion is a bit rash, don't you think?

**Pull request closed without merging**